pysph/examples/db_3d_yeh.py

```diff
@@ -0,0 +1,165 @@
+"""Dam break flow against a tall structure
+
+The case is based of an experiment done by Yeh and Petroff,
+University of Washington.
+Refer to "Peter E. Raad, Razvan Bidoae, The three-dimensional
+Eulerian–Lagrangian marker and micro cell method
+for the simulation of free surface flows, Journal of Computational Physics,
+Volume 203, Issue 2, 2005, Pages 668-699,
+ISSN 0021-9991"
+
+DOI: https://doi.org/10.1016/j.jcp.2004.09.013
+
+"""
+
+import numpy as np
+
+from pysph.base.utils import get_particle_array
+from pysph.tools.geometry import remove_overlap_particles
+from pysph.base.kernels import Gaussian
+from pysph.examples._db_geometry import DamBreak3DGeometry
+from pysph.solver.application import Application
+from pysph.sph.integrator import EPECIntegrator
+from pysph.sph.scheme import WCSPHScheme
+
+dim = 3
+
+tf = 1.4
+
+H = 0.3
+dx = H/20
+nboundary_layers = 1
+hdx = 1.32
+ro = 1000.0
+h0 = dx * hdx
+gamma = 7.0
+alpha = 0.02
+beta = 0.0
+
+c0 = 16.0 * np.sqrt(9.81 * H)
+
+
+class DamBreak3D(Application):
+    def add_user_options(self, group):
+        group.add_argument(
+            '--dx', action='store', type=float, dest='dx',  default=dx,
+            help='Particle spacing.'
+        )
+        group.add_argument(
+            '--hdx', action='store', type=float, dest='hdx',  default=hdx,
+            help='Specify the hdx factor where h = hdx * dx.'
+        )
+
+    def consume_user_options(self):
+        dx = self.options.dx
+        self.dx = dx
+        self.hdx = self.options.hdx
+        self.h0 = self.hdx * self.dx
+        self.geom = DamBreak3DGeometry(
+            container_height=0.4, container_width=0.61, container_length=1.6,
+            fluid_column_height=0.3, fluid_column_width=0.61, fluid_column_length=0.4,
+            obstacle_center_x=0.96, obstacle_center_y=0,
+            obstacle_length=0.12, obstacle_height=0.75, obstacle_width=0.12,
+            nboundary_layers=2, with_obstacle=True, dx=dx, hdx=hdx, rho0=ro
+        )
+        self.co = c0
+
+    def create_scheme(self):
+        s = WCSPHScheme(
+            ['fluid'], ['boundary', 'obstacle'], dim=dim, rho0=ro, c0=c0,
+            h0=h0, hdx=hdx, gz=-9.81, alpha=alpha, beta=beta, gamma=gamma,
+            hg_correction=True, delta_sph=True, delta=0.1
+        )
+        return s
+
+    def configure_scheme(self):
+        s = self.scheme
+        hdx = self.hdx
+        kernel = Gaussian(dim=dim)
```

Comment:
Why is Gaussian kernel the default? Could we not use cubic/quintic spline?

Reply:
δ-SPH model for simulating violent impact flows (Marrone et al., 2011) has implemented the same experiment using a δ-SPH Scheme and a Gaussian kernel. Shall I change it to Cubic Spline?

Reply:
Oh was not aware of it. Yes you can use the same kernel as the original paper or change it if you find Cubic spline better than Gaussian.